hyperhtml.js

```diff
@@ -60,129 +47,105 @@ var hyperHTML = (function () {'use strict';
-        // as long as the right attribute is addressed
-        if (IE) attribute = node.getAttributeNode(IEAttributes.shift());
-        actions.push(remapping ?
-          {a:'attr', n:attribute} :
-          setAttribute(node, attribute));
-      }
+      aLength = a.length,
+      bLength = b.length
+    ;
+    while (i < aLength) {
+      if (i < bLength && a[i] === b[i]) i++;
+      else return i;
     }
+    return i === bLength ? -1 : i;
   }
 
-  // traverse the whole node in search of editable content
-  // decide what each future update should change
-  //
-  // <div atr="${some.attribute}">
-  //    <h1>${some.HTML}</h1>
-  //    <p> ${some.text} </p>
-  // </div>
-  function lukeTreeWalker(node, actions) {
-    for (var
-      child, text,
-      childNodes = slice.call(node.childNodes),
-      length = childNodes.length,
-      i = 0; i < length; i++
-    ) {
-      child = childNodes[i];
-      switch (child.nodeType) {
-        case ELEMENT_NODE:
-          attributesSeeker(child, actions);
-          lukeTreeWalker(child, actions);
-          break;
-        case COMMENT_NODE:
-          if (child.textContent === uid) {
-            if (length === 1) {
-              if (remapping) {
-                actions.push({a:'any', n:node});
-              } else {
-                actions.push(setAnyContent(node));
-                node.removeChild(child);
-              }
-            } else if (
-              (i < 1 || childNodes[i - 1].nodeType === ELEMENT_NODE) &&
-              (i + 1 === length || childNodes[i + 1].nodeType === ELEMENT_NODE)
-            ) {
-              actions.push(remapping ?
-                {a:'vc', n:child} : setVirtualContent(child));
-            } else {
-              if (remapping) {
-                actions.push({a:'text', n:child});
-              } else {
-                text = node.ownerDocument.createTextNode('');
-                actions.push(setTextContent(text));
-                node.replaceChild(text, child);
-              }
-            }
-          }
-          break;
-        case TEXT_NODE:
-          // TODO: once SHOULD_USE_ATTRIBUTE contains more attributes
-          //       it's probably a good idea not to use it in here.
-          if (SHOULD_USE_ATTRIBUTE.test(node.nodeName) && child.textContent === uidc) {
-            actions.push(remapping ? {a:'text', n:node} : setTextContent(node));
-          }
+  // accordingly with the content type
+  // it replace the content of a node
+  // with the give child
+  function populateNode(parent, child) {
+    switch (child.nodeType) {
+      case ELEMENT_NODE:
+        var childNodes = parent.childNodes;
+        if (childNodes[0] === child) {
+          removeNodeList(childNodes, 1);
           break;
-      }
+        }
+        resetAndPopulate(parent, child);
+        break;
+      case DOCUMENT_FRAGMENT_NODE:
+        if (-1 < indexOfDifferences(parent.childNodes, child.childNodes)) {
```

Comment:
actually I've just realized I've lost many changes ... something went wrong with the merge